Firmware/MotorControl/utils.c

```diff
@@ -183,3 +184,23 @@ uint32_t timeout_to_deadline(uint32_t timeout_ms) {
     uint32_t now_ms = (uint32_t)((1000ull * (uint64_t)osKernelSysTick()) / osKernelSysTickFrequency);
     return now_ms + timeout_ms;
 }
+
+// @brief: Returns number of microseconds since system startup
+uint32_t micros(void) {
+    uint32_t usTicks = HAL_RCC_GetSysClockFreq() / 1000000;
+    register uint32_t ms, cycle_cnt;
+    do {
+        ms = HAL_GetTick();
+        cycle_cnt = SysTick->VAL;
```

Comment:
@qjones81 We don't use SysTick as timebase, we use TIM14. See the CubeMX project under SYS periperhal and/or `HAL_TIM_PeriodElapsedCallback` in main.c.
We should check if the autogenerated code somehow already emits a define or something like that for the timebase, otherwise we need to manually do it ourselves in main.h or something like that, such that this code can be easily maintained should we decide to switch the HAL timbase to another timer.